src/github.ts

```diff
@@ -2321,6 +2297,101 @@ export class GitHub {
   invalidateFileCache() {
     this.fileCache = new RepositoryFileCache(this.octokit, this.repository);
   }
+
+  private async queryPullRequestId(
+    pullRequestNumber: number
+  ): Promise<string | undefined> {
+    const query = `query pullRequestId($owner: String!, $repo: String!, $pullRequestNumber: Int!) {
+        repository(name: $repo, owner: $owner) {
+          pullRequest(number: $pullRequestNumber) {
+            id
+          }
+        }
+      }`;
+    const response = await this.graphqlRequest({
+      query,
+      owner: this.repository.owner,
+      repo: this.repository.repo,
+      pullRequestNumber,
+    });
+
+    this.logger.debug(response, 'queryPullrequestId');
+
+    const id = response?.repository?.pullRequest?.id;
+    if (id) {
+      return id as string;
+    }
+    return undefined;
+  }
+
+  private async mutatePullRequestEnableAutoMerge(
+    pullRequestId: string,
+    mergeMethod: MergeMethod
+  ) {
+    const mutation = `mutation mutateEnableAutoMerge($pullRequestId: ID!, $mergeMethod: PullRequestMergeMethod) {
+      enablePullRequestAutoMerge(
+        input: {pullRequestId: $pullRequestId, mergeMethod: $mergeMethod}
+      ) {
+        pullRequest {
+          autoMergeRequest{
+            authorEmail,
+            commitBody,
+            commitHeadline,
+            enabledAt,
+            enabledBy {
+              login
+            },
+            mergeMethod,
+            pullRequest{
+              id
+            }
+          }
+        }
+      }
+    }`;
+    const response = await this.graphqlRequest({
+      query: mutation,
+      pullRequestId,
+      mergeMethod: mergeMethod.toUpperCase(),
+    });
+
+    this.logger.debug({response}, 'mutatePullrequestEnableAutoMerge');
+  }
+
+  private async enablePullRequestAutoMerge(
+    pullRequestNumber: number,
+    mergeMethod: MergeMethod
+  ) {
+    this.logger.debug('Enable PR auto-merge');
+    const prId = await this.queryPullRequestId(pullRequestNumber);
+    if (!prId) {
+      throw new Error(`No id found for pull request ${pullRequestNumber}`);
+    }
+    try {
+      await this.mutatePullRequestEnableAutoMerge(prId, mergeMethod);
+    } catch (e: unknown) {
+      if (
+        isOctokitGraphqlResponseError(e) &&
+        (e.errors || []).find(
+          err =>
+            err.type === 'UNPROCESSABLE' &&
+            err.message.includes('Pull request is in clean status')
+        )
+      ) {
+        this.logger.debug(
+          'PR can be merged directly, do it instead of via GitHub auto-merge'
+        );
+        this.octokit.pulls.merge({
```

Comment:
Isn't this at a time that the PR is just created, and probably not ready to merge?

Reply:
If we get the error "Pull request in clean state" that's the way for GitHub to say "it is ready to be merged". That's the only situation where we merge directly.